libcontainer/cgroups/fs/memory.go

```diff
@@ -35,13 +40,10 @@ func (s *MemoryGroup) Apply(d *data) (err error) {
 		}
 	}()
 
-	if err := s.Set(path, d.c); err != nil {
-		return err
-	}
-
 	// We need to join memory cgroup after set memory limits, because
 	// kmem.limit_in_bytes can only be set when the cgroup is empty.
-	if _, err = d.join("memory"); err != nil {
+	_, err = d.join("memory")
```

Comment:
It is possible we mkdir twice, but I don't think that hurts, use `join` to enter a cgroup is simple and clear.
We can avoid this redundant, I'm not sure it's worth doing.